src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -41,9 +41,14 @@
      * login page directly), this returns the URL the user should be redirected
      * to after logging in successfully (e.g. your homepage).
      *
+     * @deprecated Implement onAuthenticationSuccess() instead of needing this function.
+     *
      * @return string
      */
-    abstract protected function getDefaultSuccessRedirectUrl();
+    protected function getDefaultSuccessRedirectUrl()
```

Comment:
Not sure I understand. Currently, the `getDefaultSuccessRedirectUrl()` method is abstract which means that nobody can use the `AbstractFormLoginAuthenticator` class without implementing it. In the wild, there is no code without a concrete implementation of this method. So, making it concrete now won't change anything as your code will *always* be overridden by user code, right?